### PR TITLE
Final

### PR DESCRIPTION
**Pull request by maryfelmendoza**
Code for review

### CODE REVIEW

**Comment by maryfelmendoza** [comment]
Please help me with the error of the stickler that I don't know how to solve it

File: code.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+module Enumerable
+  def my_each
+    for i in 0...self.length
+      yield (self[i])
```
Comment (bolah2009) [inline code.rb]: Let's remove the space beyween `yesild` and the parenthesis.

```suggestion
      yield(self[i])
```

File: code.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    end
+  end
+
+  def my_each_with_index
+    for i in 0...self.length
+      yield (self[i], i)
```
Comment (bolah2009) [inline code.rb]: Same with this, let's remove the space.

```suggestion
      yield(self[i], i)
```

Note that space is the reason you having issues with the linter.

**Review by bolah2009 — CHANGES_REQUESTED** [review]
Good start @maryfelmendoza :smile: 

See below comment to fix the linter error also test out the code with [the examples here](https://ruby-doc.org/core-2.6.4/Enumerable.html) and submit another CRR

#### :recycle:  Make these changes and submit another code review request. :+1: 

#### Feel free to reach me on slack if you having issues implementing these features. :+1: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Comment by maryfelmendoza** [comment]
The changes were made but now I have problems with self

File: code.rb [hunk 3]
```diff
@@ -1,13 +1,13 @@
 module Enumerable
   def my_each
     for i in 0...self.length
-      yield (self[i])
+      yield(self[i])
```
Comment (samgaco) [inline code.rb]: :100:

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @maryfelmendoza :high_brightness:

I see some of the changes suggested in the first review are not yet implemented:

- [ ] Please correct all the linter failing checks, you can use [rubocop](https://github.com/rubocop-hq/rubocop) in your terminal:

```
rubocop -a
```

Please, follow the requirement given by @bolah2009  


> See below comment to fix the linter error also test out the code with [the examples here](https://ruby-doc.org/core-2.6.4/Enumerable.html) and submit another CRR


Submit back for another code review once you are through those two points

----------------------------

* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

File: code.rb (hunk 3 above)
Comment (maryfelmendoza) [reply code.rb]: I made some of the changes suggested but there are other that I don't know how to resolve,
For example in the methods my_each and my_each_with_the_index the stickler says that I should replace for by each but if what I'm trying to do it, is not use each.
And about the erros of self, I dont how to resolve it too

Comment (bolah2009) [reply code.rb]: You can use other loop methods, have you tried `while` loop? :thinking:

Comment (maryfelmendoza) [reply code.rb]: Thanks a lot! Changes made!

File: code.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    while i < self.length { |i| yield(self[i]) }
```
Comment (bolah2009) [inline code.rb]: Did you test this method? :thinking: 

You seem not to be using `while` the right way. 
Also, note that in some cases you don't need to call `self`

As suggested earlier, please test the methods with [these examples](https://ruby-doc.org/core-2.6.4/Enumerable.html)

This method should also return `Enumerator` when no block is given.
:bulb: investigate the `to_enum` and `block_given?` methods.

File: code.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  end
+
+  def my_each_with_index
+    while i < self.length { |i| yield(self[i], i) }
```
Comment (bolah2009) [inline code.rb]: See the above comments on using `while` and returning `Enumerator` when no block is given.

See ruby doc on [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index) behavior.

File: code.rb [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  end
+
+  def my_select
+    result = []
+    self.my_each { |i| result.push(i) if yield(i) }
+    result
+  end
```
Comment (bolah2009) [inline code.rb]: This too as in the comment for `my_each` on returning `Enumerator` when no block is given.

See ruby doc on [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select) behavior.

**Review by bolah2009 — CHANGES_REQUESTED** [review]
Hi @maryfelmendoza 

Please test your methods before submitting a code review request. :blush: 
Currently, the `code.rb` file is not running, there's a syntax error - `unexpected end-of-input, expecting end`. Please check the comments below, you seems to be using `while` loop without `end`.

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
Hi @maryfelmendoza good job so far, I'll try my best to be very descriptive with regards to what is needed for this project to pass

**my_each**
- returns an Enumerator if no block is given

**my_each_with_index**
- returns an enumerator if no block is given

**my_select**
- my_select returns an enumerator if no block is given

**my_all, my_any, my_none**
work with a class, regex, block.

**my_count**
- takes an argument
- takes a block

**my_map**
- my_map returns an Enumerator if no block is given

**my_inject**
- combines all elements of enum by applying a binary operation, specified by a block
- when a symbol is specified, combines each element of the collection by applying the symbol as a named method

Please make changes and request another re-view :+1:

**Comment by maryfelmendoza** [comment]
Before continue I would like that you check the block given thing, please let me know if I understood well what you requested me

**Review by stratospherique — CHANGES_REQUESTED** [review]
> Before continue I would like that you check the block given thing, please let me know if I understood well what you requested me

- If no block was given, you shouldn't return a string message as you did.
- You should return an enumerator instead :nerd_face: 
:pray: Fix it

Happy coding :v: 
##### Ahmed Mahfoudh

File: code.rb [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    self
+  end
+
+  def my_select
+    return(:my_select) unless block_given?
+
+    result = []
+    my_each { |i| result.push(i) if yield(i) }
+    result
+  end
```
Comment (maelfosso) [inline code.rb]: Should return an **enumerator**
```ruby
irb(main):011:0> array
=> [6, 7, 5, 3, 7, 3, 4, 1, 7, 3, 8, 7, 8, 1, 0, 0, 3, 6, 6, 3, 0, 7, 2, 6, 6, 6, 0, 5, 5, 5, 4, 0, 3, 8, 7, 6, 8, 1, 7, 1, 7, 5, 6, 7, 8, 3, 4, 2, 0, 1, 5, 2, 7, 0, 1, 7, 0, 8, 1, 2, 4, 5, 3, 3, 6, 5, 0, 5, 1, 5, 3, 1, 7, 0, 0, 4, 0, 5, 4, 0, 0, 2, 2, 4, 8, 0, 6, 0, 3, 3, 5, 8, 8, 8, 1, 7, 7, 5, 8, 7]
irb(main):012:0> array.select
=> #<Enumerator: [6, 7, 5, 3, 7, 3, 4, 1, 7, 3, 8, 7, 8, 1, 0, 0, 3, 6, 6, 3, 0, 7, 2, 6, 6, 6, 0, 5, 5, 5, 4, 0, 3, 8, 7, 6, 8, 1, 7, 1, 7, 5, 6, 7, 8, 3, 4, 2, 0, 1, 5, 2, 7, 0, 1, 7, 0, 8, 1, 2, 4, 5, 3, 3, 6, 5, 0, 5, 1, 5, 3, 1, 7, 0, 0, 4, 0, 5, 4, 0, 0, 2, 2, 4, 8, 0, 6, 0, 3, 3, 5, 8, 8, 8, 1, 7, 7, 5, 8, 7]:select>
irb(main):013:0> array.my_select
=> :my_select
irb(main):014:0>
```

File: code.rb [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+      my_each { |i| return false unless yield(i) }
+    elsif pattern
+      my_each do |i|
+        return false unless pattern_match?(i, pattern)
+      end
+    else
+      my_each { |i| return false unless i }
+    end
+    true
+  end
```
Comment (maelfosso) [inline code.rb]:
```ruby
irb(main):016:0> array
=> [6, 7, 5, 3, 7, 3, 4, 1, 7, 3, 8, 7, 8, 1, 0, 0, 3, 6, 6, 3, 0, 7, 2, 6, 6, 6, 0, 5, 5, 5, 4, 0, 3, 8, 7, 6, 8, 1, 7, 1, 7, 5, 6, 7, 8, 3, 4, 2, 0, 1, 5, 2, 7, 0, 1, 7, 0, 8, 1, 2, 4, 5, 3, 3, 6, 5, 0, 5, 1, 5, 3, 1, 7, 0, 0, 4, 0, 5, 4, 0, 0, 2, 2, 4, 8, 0, 6, 0, 3, 3, 5, 8, 8, 8, 1, 7, 7, 5, 8, 7]
irb(main):017:0> array.all?(Integer)
=> true
irb(main):018:0> array.my_all?(Integer)
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):18
        3: from code.rb:39:in `my_all?'
        2: from code.rb:10:in `my_each'
        1: from code.rb:40:in `block in my_all?'
NoMethodError (undefined method `pattern_match?' for #<Array:0x0000558fcd184450>)
irb(main):019:0>
```

Comment (maelfosso) [reply code.rb]:
```ruby
irb(main):021:0> words = %w[dog door rod blade]
=> ["dog", "door", "rod", "blade"]
irb(main):022:0> words.all?(/d/)
=> true
irb(main):023:0> words.my_all?(/d/)
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):23
        3: from code.rb:39:in `my_all?'
        2: from code.rb:10:in `my_each'
        1: from code.rb:40:in `block in my_all?'
NoMethodError (undefined method `pattern_match?' for ["dog", "door", "rod", "blade"]:Array)
irb(main):024:0> 
```

Comment (maelfosso) [reply code.rb]:
```ruby
irb(main):026:0> array
=> [6, 7, 5, 3, 7, 3, 4, 1, 7, 3, 8, 7, 8, 1, 0, 0, 3, 6, 6, 3, 0, 7, 2, 6, 6, 6, 0, 5, 5, 5, 4, 0, 3, 8, 7, 6, 8, 1, 7, 1, 7, 5, 6, 7, 8, 3, 4, 2, 0, 1, 5, 2, 7, 0, 1, 7, 0, 8, 1, 2, 4, 5, 3, 3, 6, 5, 0, 5, 1, 5, 3, 1, 7, 0, 0, 4, 0, 5, 4, 0, 0, 2, 2, 4, 8, 0, 6, 0, 3, 3, 5, 8, 8, 8, 1, 7, 7, 5, 8, 7]
irb(main):027:0> array.all?(3)
=> false
irb(main):028:0> array.my_all?(3)
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):28
        3: from code.rb:39:in `my_all?'
        2: from code.rb:10:in `my_each'
        1: from code.rb:40:in `block in my_all?'
NoMethodError (undefined method `pattern_match?' for #<Array:0x0000558fcd184450>)
irb(main):029:0>
```

File: code.rb [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+      my_each { |i| return true if yield(i) }
+    elsif pattern
+      my_each do |i|
+        return true if pattern_match?(i, pattern)
+      end
+    else
+      my_each { |i| return true if i }
+    end
+    false
+  end
```
Comment (maelfosso) [inline code.rb]:
```ruby
irb(main):032:0> array
=> [6, 7, 5, 3, 7, 3, 4, 1, 7, 3, 8, 7, 8, 1, 0, 0, 3, 6, 6, 3, 0, 7, 2, 6, 6, 6, 0, 5, 5, 5, 4, 0, 3, 8, 7, 6, 8, 1, 7, 1, 7, 5, 6, 7, 8, 3, 4, 2, 0, 1, 5, 2, 7, 0, 1, 7, 0, 8, 1, 2, 4, 5, 3, 3, 6, 5, 0, 5, 1, 5, 3, 1, 7, 0, 0, 4, 0, 5, 4, 0, 0, 2, 2, 4, 8, 0, 6, 0, 3, 3, 5, 8, 8, 8, 1, 7, 7, 5, 8, 7]
irb(main):033:0> array.any?(Integer)
=> true
irb(main):034:0> array.my_any?(Integer)
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):34
        3: from code.rb:52:in `my_any?'
        2: from code.rb:10:in `my_each'
        1: from code.rb:53:in `block in my_any?'
NoMethodError (undefined method `pattern_match?' for #<Array:0x0000558fcd184450>)
irb(main):035:0>
```

Comment (maelfosso) [reply code.rb]:
```ruby
irb(main):037:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):038:0> words.any?(/z/)
=> false
irb(main):039:0> words.my_any?(/z/)
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):39
        3: from code.rb:52:in `my_any?'
        2: from code.rb:10:in `my_each'
        1: from code.rb:53:in `block in my_any?'
NoMethodError (undefined method `pattern_match?' for ["dog", "door", "rod", "blade"]:Array)
irb(main):040:0> 
```

Comment (maelfosso) [reply code.rb]:
```ruby
irb(main):042:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):043:0> words.any?('cat')
=> false
irb(main):044:0> words.my_any?('cat')
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):44
        3: from code.rb:52:in `my_any?'
        2: from code.rb:10:in `my_each'
        1: from code.rb:53:in `block in my_any?'
NoMethodError (undefined method `pattern_match?' for ["dog", "door", "rod", "blade"]:Array)
irb(main):045:0>
```

File: code.rb [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+      my_each { |i| return false if yield(i) }
+    elsif pattern
+      my_each do |i|
+        return false if pattern_match?(i, pattern)
+      end
+    else
+      my_each { |i| return false if i }
+    end
+    true
+  end
```
Comment (maelfosso) [inline code.rb]:
```ruby
irb(main):047:0> array
=> [6, 7, 5, 3, 7, 3, 4, 1, 7, 3, 8, 7, 8, 1, 0, 0, 3, 6, 6, 3, 0, 7, 2, 6, 6, 6, 0, 5, 5, 5, 4, 0, 3, 8, 7, 6, 8, 1, 7, 1, 7, 5, 6, 7, 8, 3, 4, 2, 0, 1, 5, 2, 7, 0, 1, 7, 0, 8, 1, 2, 4, 5, 3, 3, 6, 5, 0, 5, 1, 5, 3, 1, 7, 0, 0, 4, 0, 5, 4, 0, 0, 2, 2, 4, 8, 0, 6, 0, 3, 3, 5, 8, 8, 8, 1, 7, 7, 5, 8, 7]
irb(main):048:0> array.none?(String)
=> true
irb(main):049:0> array.my_none?(String)
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):49
        3: from code.rb:65:in `my_none?'
        2: from code.rb:10:in `my_each'
        1: from code.rb:66:in `block in my_none?'
NoMethodError (undefined method `pattern_match?' for #<Array:0x0000558fcd184450>)
irb(main):050:0>
```

Comment (maelfosso) [reply code.rb]:
```ruby
irb(main):052:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):053:0> words.none?(/z/)
=> true
irb(main):054:0> words.my_none?(/z/)
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):54
        3: from code.rb:65:in `my_none?'
        2: from code.rb:10:in `my_each'
        1: from code.rb:66:in `block in my_none?'
NoMethodError (undefined method `pattern_match?' for ["dog", "door", "rod", "blade"]:Array)
irb(main):055:0>
```

Comment (maelfosso) [reply code.rb]:
```ruby
irb(main):057:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):058:0> words.none?('dog')
=> false
irb(main):059:0> words.my_none?('dog')
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):59
        3: from code.rb:65:in `my_none?'
        2: from code.rb:10:in `my_each'
        1: from code.rb:66:in `block in my_none?'
NoMethodError (undefined method `pattern_match?' for ["dog", "door", "rod", "blade"]:Array)
irb(main):060:0> 
```

File: code.rb [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+
+  def my_count
+    my_select { |x| yield(x) }.size
+  end
```
Comment (maelfosso) [inline code.rb:112]:
```ruby
irb(main):064:0> array
=> [6, 7, 5, 3, 7, 3, 4, 1, 7, 3, 8, 7, 8, 1, 0, 0, 3, 6, 6, 3, 0, 7, 2, 6, 6, 6, 0, 5, 5, 5, 4, 0, 3, 8, 7, 6, 8, 1, 7, 1, 7, 5, 6, 7, 8, 3, 4, 2, 0, 1, 5, 2, 7, 0, 1, 7, 0, 8, 1, 2, 4, 5, 3, 3, 6, 5, 0, 5, 1, 5, 3, 1, 7, 0, 0, 4, 0, 5, 4, 0, 0, 2, 2, 4, 8, 0, 6, 0, 3, 3, 5, 8, 8, 8, 1, 7, 7, 5, 8, 7]
irb(main):065:0> array.count
=> 100
irb(main):066:0> array.my_count
Traceback (most recent call last):
        7: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        6: from (irb):66
        5: from code.rb:75:in `my_count'
        4: from code.rb:31:in `my_select'
        3: from code.rb:10:in `my_each'
        2: from code.rb:31:in `block in my_select'
        1: from code.rb:75:in `block in my_count'
LocalJumpError (no block given (yield))
irb(main):067:0>
```

Comment (maelfosso) [reply code.rb:112]:
```ruby
irb(main):069:0> array
=> [6, 7, 5, 3, 7, 3, 4, 1, 7, 3, 8, 7, 8, 1, 0, 0, 3, 6, 6, 3, 0, 7, 2, 6, 6, 6, 0, 5, 5, 5, 4, 0, 3, 8, 7, 6, 8, 1, 7, 1, 7, 5, 6, 7, 8, 3, 4, 2, 0, 1, 5, 2, 7, 0, 1, 7, 0, 8, 1, 2, 4, 5, 3, 3, 6, 5, 0, 5, 1, 5, 3, 1, 7, 0, 0, 4, 0, 5, 4, 0, 0, 2, 2, 4, 8, 0, 6, 0, 3, 3, 5, 8, 8, 8, 1, 7, 7, 5, 8, 7]
irb(main):070:0> array.count(3)
=> 12
irb(main):071:0> array.my_count(3)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):71
        1: from code.rb:74:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):072:0> 
```

File: code.rb [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+  def my_map(&proc)
+    arr = []
+    my_each { |i| arr << proc.call(i) }
+    arr
+  end
```
Comment (maelfosso) [inline code.rb]: Should return an **enumerator**. Use **to_enum**
```ruby
irb(main):074:0> array
=> [6, 7, 5, 3, 7, 3, 4, 1, 7, 3, 8, 7, 8, 1, 0, 0, 3, 6, 6, 3, 0, 7, 2, 6, 6, 6, 0, 5, 5, 5, 4, 0, 3, 8, 7, 6, 8, 1, 7, 1, 7, 5, 6, 7, 8, 3, 4, 2, 0, 1, 5, 2, 7, 0, 1, 7, 0, 8, 1, 2, 4, 5, 3, 3, 6, 5, 0, 5, 1, 5, 3, 1, 7, 0, 0, 4, 0, 5, 4, 0, 0, 2, 2, 4, 8, 0, 6, 0, 3, 3, 5, 8, 8, 8, 1, 7, 7, 5, 8, 7]
irb(main):075:0> array.map
=> #<Enumerator: [6, 7, 5, 3, 7, 3, 4, 1, 7, 3, 8, 7, 8, 1, 0, 0, 3, 6, 6, 3, 0, 7, 2, 6, 6, 6, 0, 5, 5, 5, 4, 0, 3, 8, 7, 6, 8, 1, 7, 1, 7, 5, 6, 7, 8, 3, 4, 2, 0, 1, 5, 2, 7, 0, 1, 7, 0, 8, 1, 2, 4, 5, 3, 3, 6, 5, 0, 5, 1, 5, 3, 1, 7, 0, 0, 4, 0, 5, 4, 0, 0, 2, 2, 4, 8, 0, 6, 0, 3, 3, 5, 8, 8, 8, 1, 7, 7, 5, 8, 7]:map>
irb(main):076:0> array.my_map
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):76
        3: from code.rb:80:in `my_map'
        2: from code.rb:10:in `my_each'
        1: from code.rb:80:in `block in my_map'
NoMethodError (undefined method `call' for nil:NilClass)
irb(main):077:0> 
```

File: code.rb [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+
+  def my_inject
+    memo = 0
+    my_each { |i| memo = yield(memo, i) }
+    memo
+  end
```
Comment (maelfosso) [inline code.rb:126]:
```ruby
irb(main):079:0> range = Range.new(5, 50)
=> 5..50
irb(main):080:0> range.inject(4) { |prod, n| prod * n }
=> 5069015533618896340602101361010794807396273594826752000000000000
irb(main):081:0> range.my_inject(4) { |prod, n| prod * n }
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):81
        1: from code.rb:84:in `my_inject'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):082:0>
```

Comment (maelfosso) [reply code.rb:126]:
```ruby
irb(main):084:0> array
=> [6, 7, 5, 3, 7, 3, 4, 1, 7, 3, 8, 7, 8, 1, 0, 0, 3, 6, 6, 3, 0, 7, 2, 6, 6, 6, 0, 5, 5, 5, 4, 0, 3, 8, 7, 6, 8, 1, 7, 1, 7, 5, 6, 7, 8, 3, 4, 2, 0, 1, 5, 2, 7, 0, 1, 7, 0, 8, 1, 2, 4, 5, 3, 3, 6, 5, 0, 5, 1, 5, 3, 1, 7, 0, 0, 4, 0, 5, 4, 0, 0, 2, 2, 4, 8, 0, 6, 0, 3, 3, 5, 8, 8, 8, 1, 7, 7, 5, 8, 7]
irb(main):085:0> array.inject(:+)
=> 404
irb(main):086:0> array.my_inject(:+)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):86
        1: from code.rb:84:in `my_inject'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):087:0> 
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required

Hello @maryfelmendoza 

Nice improvements :clap: The result is awesome. 
However there are some bugs you have to fix. One of the main errors is the missing method **patter_match?**. Read the comments below, to know more about the errors.

Kindly make the changes and ensure you that your code passed at least the test in comment then submit a new code review request.
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

File: code.rb [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+    else
+
+      my_each do |element|
+        none = false if element
+      end
+
+    end
+
+    none
+  end
```
Comment (maelfosso) [inline code.rb:108]:
```ruby
irb(main):038:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):039:0> words.none?(/d/)
=> false
irb(main):040:0> words.my_none?(/d/)
=> true
irb(main):041:0>
```

File: code.rb [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+    else
+
+      my_each do |element|
+        any = true if element
+      end
+
+    end
+
+    any
+  end
```
Comment (maelfosso) [inline code.rb:83]:
```ruby
irb(main):023:0> array
=> [8, 7, 8, 5, 4, 0, 0, 2, 6, 3, 0, 6, 5, 4, 2, 8, 0, 7, 6, 7, 6, 1, 0, 7, 6, 0, 8, 6, 4, 6, 1, 1, 8, 7, 0, 8, 5, 1, 7, 6, 1, 5, 3, 3, 7, 5, 6, 6, 4, 6, 5, 2, 4, 3, 8, 3, 7, 1, 8, 3, 3, 0, 0, 7, 3, 6, 6, 1, 4, 4, 5, 1, 6, 1, 5, 4, 0, 0, 5, 8, 6, 1, 5, 0, 6, 1, 4, 1, 3, 6, 6, 7, 0, 5, 3, 4, 2, 7, 6, 4]
irb(main):024:0> array.any?(Integer)
=> true
irb(main):025:0> array.my_any?(Integer)
=> false
irb(main):026:0>
```

Comment (maelfosso) [reply code.rb:83]:
```ruby
irb(main):028:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):029:0> words.any?(/d/)
=> true
irb(main):030:0> words.my_any?(/d/)
=> false
irb(main):031:0> 
```

File: code.rb [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+    else
+
+      my_each do |element|
+        all_true = false unless element
+      end
+
+    end
+
+    all_true
+  end
```
Comment (maelfosso) [inline code.rb:59]:
```ruby
irb(main):013:0> array
=> [8, 7, 8, 5, 4, 0, 0, 2, 6, 3, 0, 6, 5, 4, 2, 8, 0, 7, 6, 7, 6, 1, 0, 7, 6, 0, 8, 6, 4, 6, 1, 1, 8, 7, 0, 8, 5, 1, 7, 6, 1, 5, 3, 3, 7, 5, 6, 6, 4, 6, 5, 2, 4, 3, 8, 3, 7, 1, 8, 3, 3, 0, 0, 7, 3, 6, 6, 1, 4, 4, 5, 1, 6, 1, 5, 4, 0, 0, 5, 8, 6, 1, 5, 0, 6, 1, 4, 1, 3, 6, 6, 7, 0, 5, 3, 4, 2, 7, 6, 4]
irb(main):014:0> array.all?(Integer)
=> true
irb(main):015:0> array.my_all?(Integer)
=> false
irb(main):016:0> 
```

Comment (maelfosso) [reply code.rb:59]:
```ruby
irb(main):018:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):019:0> words.all?(/d/)
=> true
irb(main):020:0> words.my_all?(/d/)
=> false
irb(main):021:0>
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required

Hello @maryfelmendoza ,

Thank you for the stickler's errors fixed !!
However there are always the others bugs. Please, fix them and with the new ones.

Kindly make the changes and submit the code review request.
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Pull request closed without merging**